src/View/Form/EntityContext.php

```diff
@@ -538,15 +538,20 @@ public function getMaxLength(string $field): ?int
         $parts = explode('.', $field);
         $validator = $this->_getValidator($parts);
         $fieldName = array_pop($parts);
-        if (!$validator->hasField($fieldName)) {
-            return null;
-        }
-        foreach ($validator->field($fieldName)->rules() as $rule) {
-            if ($rule->get('rule') === 'maxLength') {
-                return $rule->get('pass')[0];
+
+        if ($validator->hasField($fieldName)) {
+            foreach ($validator->field($fieldName)->rules() as $rule) {
+                if ($rule->get('rule') === 'maxLength') {
+                    return $rule->get('pass')[0];
+                }
             }
         }
 
+        $attributes = $this->attributes($field);
+        if (!empty($attributes['length'])) {
```

Comment:
If the length is 0 should we return it?

Reply:
I don't think db columns with length `0` or `maxlength` attribute with `'0'` has any practical use case, so such a default doesn't make much sense.